Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm

```diff
@@ -736,6 +741,15 @@ static bool hasValue(const ImageBufferBackendHandle& backendHandle)
 #endif
 
     layer.contents = contents.get();
+    if ([CALayer instancesRespondToSelector:@selector(contentsDirtyRect)]) {
+        if (m_paintedRect) {
+            CGRect dirty = *m_paintedRect;
+            FloatRect painted = *m_paintedRect;
+            painted.scale(layer.contentsScale);
```

Comment:
Is this right if the layer has non-zero bounds.origin? I'm not sure we ever have layers with painted content and non-zero bounds origin (we only use bounds origin on scrolling layers).

Reply:
I believe so, CA wants a rect in texture-coords, and that's what our dirty rect should be.